File: android/src/main/java/ai/xmax/reactnative/XmaxImageVideoFormat.kt
package ai.xmax.reactnative

/** Checks native frame values; MediaService owns model buckets and pixel bounds. */
internal object XmaxImageVideoFormat {
  /** Dimensions must fit the RTC frame's signed 32-bit fields. */
  fun isValid(width: Double, height: Double, fps: Double): Boolean =
    listOf(width, height).all {
      it.isFinite() && it > 0 && it % 1 == 0.0 && it <= Int.MAX_VALUE.toDouble()
    } && fps.isFinite() && fps > 0 && fps % 1 == 0.0 && fps <= 60
}

File: ios/XmaxImageVideoFormat.swift
/// Validates native frame dimensions without duplicating model resolution rules.
enum XmaxImageVideoFormat {
  /// Model buckets and pixel bounds have already been resolved by MediaService.
  /// Dimensions must fit the RTC frame's signed 32-bit fields.
  static func isValid(width: Double, height: Double, fps: Double) -> Bool {
    [width, height].allSatisfy {
      $0.isFinite && $0 > 0 && $0.rounded(.down) == $0 && $0 <= Double(Int32.max)
    } && fps.isFinite && fps > 0 && fps.rounded(.down) == fps && fps <= 60
  }
}
